File: catalogue-api/src/test/java/io/quartic/catalogue/api/WebsocketDatasetLocatorShould.java
package io.quartic.catalogue.api;

public class WebsocketDatasetLocatorShould extends DatasetLocatorTests<WebsocketDatasetLocator> {
    @Override
    protected WebsocketDatasetLocator locator() {
        return WebsocketDatasetLocatorImpl.of("foo");
    }

    @Override
    protected String json() {
        return "{\"type\": \"websocket\", \"url\": \"foo\"}";
    }
}

File: catalogue-api/src/test/kotlin/io/quartic/catalogue/api/WebsocketDatasetLocatorShould.kt
package io.quartic.catalogue.api

class WebsocketDatasetLocatorShould : DatasetLocatorTests<WebsocketDatasetLocator>() {
    override fun locator() = WebsocketDatasetLocator("foo")

    override fun json() = "{\"type\": \"websocket\", \"url\": \"foo\"}"
}
